### PR TITLE
refactor: update crate dependencies

### PR DESCRIPTION
**Pull request by didier-wenzek**
## Proposed changes

- [x] cargo update
- [x] fix mockito-based tests broken after dependencies update
- [x] fix actor runtime test broken after dependencies update

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
<br/>

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [x] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: crates/extensions/c8y_auth_proxy/src/server.rs [hunk 1]
```diff
@@ -841,7 +841,7 @@ mod tests {
     #[tokio::test]
     async fn forwards_successful_responses() {
         let _ = env_logger::try_init();
-        let mut server = mockito::Server::new();
+        let mut server = mockito::Server::new_async().await;
         let _mock = server
```
Comment (rina23q) [inline crates/extensions/c8y_auth_proxy/src/server.rs:845]: Don't we need to change from `crete()` to `create_async()` in this file?

File: crates/extensions/tedge_downloader_ext/src/tests.rs [hunk 2]
```diff
@@ -12,7 +12,7 @@ const TEST_TIMEOUT: Duration = Duration::from_secs(5);
 #[tokio::test]
 async fn download_without_auth() {
     let ttd = TempTedgeDir::new();
-    let mut server = mockito::Server::new();
+    let mut server = mockito::Server::new_async().await;
```
Comment (rina23q) [inline crates/extensions/tedge_downloader_ext/src/tests.rs:15]: The same comment as `extensions/c8y_auth_proxy/src/server.rs`.
Don't we need to change from crete() to create_async() in this file?

**Review by rina23q — COMMENTED** [review]
`create()` was not replaced by `create_async()` in two files.

File: crates/extensions/c8y_auth_proxy/src/server.rs (hunk 1 above)
Comment (Bravo555) [reply crates/extensions/c8y_auth_proxy/src/server.rs:845]: Looking at the crate documentation, there doesn't seem to be much difference between the two, i.e. both will work here, but using `.create_async` we don't block and await, which would maybe have performance implications when registering many mocks, but in our use case there's no difference as long as `.create` also works

File: crates/extensions/tedge_downloader_ext/src/tests.rs (hunk 2 above)
Comment (Bravo555) [reply crates/extensions/tedge_downloader_ext/src/tests.rs:15]: https://github.com/thin-edge/thin-edge.io/pull/3216#discussion_r1822733495

File: crates/core/tedge_actors/src/runtime.rs [hunk 3]
```diff
@@ -589,7 +589,7 @@ mod tests {
             .expect("Actor to panic in time");
         assert_eq!(
             error.map(|s| s.replace(char::is_numeric, "")), // ignore the task id
-            Some("task  panicked".to_string())
+            Some("task  panicked with message \"Oh dear\"".to_string())
```
Comment (Bravo555) [inline crates/core/tedge_actors/src/runtime.rs:592]: typo: there's two spaces in `task  panicked`

File: crates/extensions/c8y_auth_proxy/src/server.rs (hunk 1 above)
Comment (didier-wenzek) [reply crates/extensions/c8y_auth_proxy/src/server.rs:845]: Thanks @Bravo555 for the insight. However, I decided to apply @rina23q suggestions to be consistent.

File: crates/extensions/c8y_auth_proxy/src/server.rs [hunk 4]
```diff
@@ -890,7 +892,7 @@ mod tests {
     #[tokio::test]
     async fn forwards_unsuccessful_responses() {
         let _ = env_logger::try_init();
-        let mut server = mockito::Server::new();
+        let mut server = mockito::Server::new_async().await;
         let _mock = server
             .mock("GET", "/not-a-known-url")
             .with_status(404)
```
Comment (rina23q) [inline crates/extensions/c8y_auth_proxy/src/server.rs:898]: You missed to apply `create_async` to the line below. (line 899)

Comment (didier-wenzek) [reply crates/extensions/c8y_auth_proxy/src/server.rs:898]: :-) I will fixed that.

That said this demonstrates that @Bravo555 comment is correct. The tests are working as expected with `create()` as well ass with `create_async()`.

**Review by rina23q — APPROVED** [review]
Approved

**Pull request merged**